src/main/java/org/apache/commons/pool3/impl/GenericObjectPool.java

```diff
@@ -513,6 +513,11 @@ private PooledObject<T> create(final Duration maxWaitDurationRequest) throws E {
         if (localMaxTotal < 0) {
             localMaxTotal = Integer.MAX_VALUE;
         }
+        int localMaxIdle = getMaxIdle();
+        if (localMaxIdle < 0) {
+            localMaxIdle = Integer.MAX_VALUE;
+        }
+        final int maxCapacity = Math.min(localMaxTotal, localMaxIdle);
```

Comment:
This will cause issues if maxIdle is set to zero, making it impossible to create new objects in the pool.

Since the method is private we can verify it is only used during the borrow, where it is not added to the idle objects, and on addObject, where it is.

the `addObject` method should be checking the idle object limit isn't being crossed.